image_classification/tensorflow/official/resnet/resnet_run_loop.py

```diff
@@ -70,12 +74,22 @@ def process_record_dataset(dataset, is_training, batch_size, shuffle_buffer,
   # dataset for the appropriate number of epochs.
   dataset = dataset.repeat(num_epochs)
 
+  if is_training and num_gpus and examples_per_epoch:
+    total_examples = num_epochs * examples_per_epoch
+    # Force the number of batches to be divisible by the number of devices.
+    # This prevents some devices from receiving batches while others do not,
+    # which can lead to a lockup. This case will soon be handled directly by
+    # distribution strategies, at which point this .take() operation will no
+    # longer be needed.
+    total_batches = total_examples // batch_size // num_gpus * num_gpus
+    dataset.take(total_batches * batch_size)
```

Comment:
Should this be 'dataset = dataset.take(total_batches * batch_size)'?

Reply:
I followed up with some internal folks, and our conclusion was that this is indeed a no-op, and the issue that it we trying to address is no longer a problem. I will remove it. Thanks for catching it.